lib/init/passport.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+/**
+ * @fileoverview Passport authentication init
+ */
+const config = require('config');
+const passport = require('passport');
+const LocalStrategy = require('passport-local').Strategy;
+const bcrypt = require('bcrypt-nodejs');
+
+const acl = require('../auth/acl');
+
+const database = require('../database');
+
+// Passport Authentication setup
+passport.use(new LocalStrategy(
+  {
+    usernameField: 'email',
+  },
+  (username, password, cb) => {
+    database.select('*').from('users').where({
+      email: username,
+    })
+    .then(user => {
+      if ((user.length < 1) || (!bcrypt.compareSync(password, user[0].password))) {
+        return cb(null, false, { message: config.authentication.messages.badlogin });
+      }
+
+      // register user's role for access
+      acl.addUserRoles(user[0].id.toString(), user[0].role);
```

Comment:
`addUserRoles`? How about `addRoles` or `roles`? Or `register` as we've got `reregister` to re-register a user's roles